hivemind/client/averaging/__init__.py

```diff
@@ -265,10 +265,10 @@ def step(self, gather: Optional[DataForGather] = None, weight: float = 1.0, time
         :param wait: if True (default), return when finished. Otherwise return MPFuture and run in background.
         :returns: on success, update averaged_tensors and return group info; on failure, return None
         """
-        if self.mode == AveragingMode.AUX and weight != 1:
+        if self.mode == AveragingMode.AUX and weight is not None:
             logger.warning("Averager is running in auxiliary mode, weight is unused.")
-        else:
-            assert isinstance(weight, (int, float)) and weight > 0, f"Expected a positive int/float, got {type(weight)}"
+        weight = weight if weight is not None else float(self.mode != AveragingMode.AUX)
+        assert isinstance(weight, (int, float)) and weight >= 0, f"Expected a positive int/float, got {type(weight)}"
```

Comment:
Maybe expect strictly a float, as suggested by the type hint?

Reply:
BTW the assertion for type is somewhat redundant if we already have type hints

Reply:
This is an intended behavior.

**(1)** typing supports this:
>Rather than requiring that users write import numbers and then use numbers.Float etc., this PEP proposes a straightforward shortcut that is almost as effective: when an argument is annotated as having type float, an argument of type int is acceptable; similar, for an argument annotated as having type complex, arguments of type float or int are acceptable.

https://www.python.org/dev/peps/pep-0484/#the-numeric-tower

**(2)** isinstance requires explicit checks:
![image](https://user-images.githubusercontent.com/3491902/123641152-9e3dac00-d82a-11eb-8f07-d58aabe028f4.png)

Reply:
Understood, but do we need this assertion for types anyway if we explicitly indicate that we need a numeric type here? Seems like excessive checking to me

Reply:
Yes, we do.